docs/CORE_DEV_GUIDE.md

```diff
@@ -0,0 +1,180 @@
+doubt, always refer back here.
+
+As a core team member, you gain the responsibility of shepherding
+other contributors through the review process; here are some
+guidelines.
+
+## All Contributors Are Treated The Same
+
+As a core developer, you gain the ability to merge or approve
+other contributors' pull requests.  Much like nuclear launch keys, it
+is a shared power: you must merge *only after* another core has
+approved the pull request, *and* after you yourself have carefully
+reviewed it.  (See `Reviewing` and especially `Merge Only Changes You
+Understand` below.) To ensure a clean git history, use GitHub's
+[Squash and Merge](https://help.github.com/articles/merging-a-pull-request/#merging-a-pull-request-on-github)
+feature to merge, unless you have a very good reason not to do so.
+
+You should continue to make your own pull requests as before and in accordance
+with the [general contributor guide](CONTRIBUTING.md). These pull requests still
+require the approval of another core developer before they can be merged.
+
+## Reviewing
+
+### How to Conduct A Good Review
+
+*Always* be kind to contributors. Contributors are often doing
+volunteer work, for which we are tremendously grateful. Provide
+constructive criticism on ideas and implementations, and remind
+yourself of how it felt when your own work was being evaluated as a
+novice.
+
+`napari` strongly values mentorship in code review.  New users
+often need more handholding, having little to no git
+experience. Repeat yourself liberally, and, if you don’t recognize a
+contributor, point them to our development guide, or other GitHub
+workflow tutorials around the web. Do not assume that they know how
+GitHub works (e.g., many don't realize that adding a commit
+automatically updates a pull request). Gentle, polite, kind
+encouragement can make the difference between a new core developer and
+an abandoned pull request.
+
+When reviewing, focus on the following:
+
+1. **Usability and generality:** `napari` is an GUI application that strives to be accessible
+to both coding and non-coding users, and new features should ultimately be
+accessible to everyone using the app. `napari` targets the scientific user
+community broadly, and core features should be domain agnostic and general purpose.
+Custom functionality is meant to be provided through our plugin ecosystem. If in doubt
+consult back with our [mission and values](MISSION_AND_VALUES.md).
+
+2. **Performance and benchmarks:** As `napari` targets scientific applications which often involve
+large multidimensional datasets, high performance is a key value of `napari`. While
+every new feature won't scale equally to all sizes of data keeping in mind performance
+and our [benchmarks](BENCHMARKS.md) during a review may be important, and you may
+need to ask for benchmarks to be run and reported or new benchmarks to be added.
+
+3. **APIs and stability:** Coding users and plugin developers will make
```

Comment:
does this mean that we want to start looking at a deprecation cycle?

Reply:
maybe soonish. think we can maybe do one more breaking `0.3` before we have too, but we might want to bundle a lot of changes into that release. we might want to start having a `dev0.0.3` branch for us to work on too

Reply:
i've said we don't have one yet, but will work to add one soon. we can discuss this, maybe we can adopt one now, maybe we delay till the next major release to discuss